src/common/dao/project.go

```diff
@@ -241,12 +284,31 @@ func DeleteProject(id int64) error {
 	if err != nil {
 		return err
 	}
-
 	name := fmt.Sprintf("%s#%d", project.Name, project.ProjectID)
-
 	sql := `update project 
-		set deleted = 1, name = ? 
-		where project_id = ?`
+		    set deleted = 1, name = ? 
+		    where project_id = ?`
 	_, err = GetOrmer().Raw(sql, name, id).Exec()
 	return err
 }
+
+// GetRolesByLDAPGroup - Get Project roles of the
+// specified group DN is a member of current project
+func GetRolesByLDAPGroup(projectID int64, groupDNCondition string) []int {
+	var roles []int
+	if len(groupDNCondition) == 0 {
+		return roles
+	}
+	o := GetOrmer()
+	sql := fmt.Sprintf(
+		`select pm.role from project_member pm 
+		left join user_group ug on pm.entity_type = 'g' and pm.entity_id = ug.id 
+		where ug.ldap_group_dn in ( %s ) and pm.project_id = ? `,
+		groupDNCondition)
+	log.Debugf("sql:%v", sql)
+	if _, err := o.Raw(sql, projectID).QueryRows(&roles); err != nil {
+		log.Warningf("Error in GetRolesByLDAPGroup, error: %v", err)
+		return nil
```

Comment:
Why don't we return the err but just log a warning message?

Reply:
Return error now